File: 30DaysOfCode/solution_04.py
# Task
# Write a Person class with an instance variable,
# and a constructor that takes an integer, as a parameter.
# The constructor must assign  to  after confirming the argument passed as
# is not negative; if a negative argument is passed as ,
# the constructor should set  to  and print Age is not valid, setting age to 0.
# In addition, you must write the following instance methods:
#
# yearPasses() should increase the  instance variable by .
# amIOld() should perform the following conditional actions:
# If , print You are young..
# If  and , print You are a teenager..
# Otherwise, print You are old..


class Person:
    def __init__(self, initialAge):
        # Add some more code to run some checks on initialAge
        if initialAge < 0:
            print("Age is not valid, setting age to 0.")
            self.age = 0
        else:
            self.age = initialAge

    def yearPasses(self):
        # Increment the age of the person in here
        self.age = self.age + 1
        return

    def amIOld(self):
        # Do computations & print correct statement to console
        # Switch statements avaialble in pyv3.10 only! Hackerrank's py is 3.9.4
        if self.age < 13:
            print("You are young.")
        elif self.age >= 13 and self.age < 18:
            print("You are a teenager.")
        else:
            print("You are old.")
        return


# Input:
t = int(input())
for i in range(0, t):
    age = int(input())
    p = Person(age)
    p.amIOld()
    for j in range(0, 3):
        p.yearPasses()
    p.amIOld()
    print("")

File: 04-ClassVsInstance/solution_04.scala
import java.io._
import java.math._
import java.security._
import java.text._
import java.util._
import java.util.concurrent._
import java.util.function._
import java.util.regex._
import java.util.stream._
import scala.collection.immutable._
import scala.collection.mutable._
import scala.collection.concurrent._
import scala.concurrent._
import scala.io._
import scala.math._
import scala.sys._
import scala.util.matching._
import scala.reflect._


class Person {

    var age: Int = 0

    def this(initialAge: Int) = {
        // Add some more code to run some checks on initialAge
        this()
        if (initialAge >= 0) this.age = initialAge
        else println("Age is not valid, setting age to 0.")
        }

    def amIOld(): Unit = {
        // Do some computations in here and print out the correct statement to the console
        if (this.age < 13) println("You are young.")
        else if (this.age >= 13 && this.age < 18) println("You are a teenager.")
        else println("You are old.")
        }

    def yearPasses(): Unit = {
        // Increment the age of the person in here
        this.age += 1
        }
}

object Solution {

    def main(args: Array[String]) = {
        var T = scala.io.StdIn.readInt()
        var i = 0
        for(i <- 1 to T){        
             var age=scala.io.StdIn.readInt()
             var p=new Person(age)
             p.amIOld()
             var j=0
             for(j <- 1 to 3){
                 p.yearPasses()
             }
             p.amIOld()
             System.out.println()

        }


    }
}